tests/static-analysis/contains.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+/**
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+include __DIR__ . '/../../vendor/autoload.php';
+
+use loophp\collection\Collection;
+use loophp\collection\Contract\Collection as CollectionInterface;
+
+/**
+ * @param CollectionInterface<int, bool> $collection
+ */
+function contains_checkList(CollectionInterface $collection): void
+{
+}
+/**
+ * @param CollectionInterface<string, bool> $collection
+ */
+function contains_checkMap(CollectionInterface $collection): void
+{
+}
+function contains_checkBool(bool $value): void
+{
+}
+
+contains_checkList(Collection::fromIterable([1, 2, 3])->contains(2));
+contains_checkList(Collection::fromIterable(range(1, 3))->contains(2, 4));
+
+contains_checkMap(Collection::fromIterable(['foo' => 'bar', 'baz' => 'taz'])->contains('bar'));
+/** @psalm-suppress InvalidArgument -> Psalm is smart and knows that 'abc' cannot be in there */
+contains_checkMap(Collection::fromIterable(['foo' => 'bar', 'baz' => 'taz'])->contains('bar', 'abc'));
+
+// VALID failures below -> `current` can return `NULL` if the collection is empty
+
+/** @psalm-suppress InvalidArgument @phpstan-ignore-next-line */
+contains_checkBool(Collection::fromIterable([1, 2, 3])->contains(2)->current());
+/** @psalm-suppress InvalidArgument @phpstan-ignore-next-line */
+contains_checkBool(Collection::fromIterable(['foo' => 'bar', 'baz' => 'taz'])->contains('bar')->current());
```

Comment:
these feel very awkward to me. `current()` returns `T|null`, with `null` being returned when the collection is empty. However, in the chaining `contains()->current()` this will never be `null` because `contains()` will always return a collection with a boolean

Reply:
I guess in practice when used like this the analysers will not complain:

```PHP
if (true === Collection::fromIterable([1, 2, 3])->contains(2)->current()) {
    // do stuff
}
```

but if the comparison is not explicit PHPStan at least will complain because the return type is `bool|null`:

```PHP
if (Collection::fromIterable([1, 2, 3])->contains(2)->current()) {
    // do stuff
}```

Reply:
@drupol I'm not sure I can convince you to make `contains` just return `bool` 😛. But it would be so much easier 😅

Reply:
It was like that in the early versions of the library then I changed. Maybe we should discuss about that in a dedicate PR ?